projects/gloo/pkg/plugins/ratelimit/plugin.go

```diff
@@ -44,6 +49,7 @@ func (p *Plugin) Init(params plugins.InitParams) error {
 		p.upstreamRef = rlServer.RatelimitServerRef
 		p.timeout = rlServer.RequestTimeout
 		p.denyOnFail = rlServer.DenyOnFail
+		p.rateLimitBeforeAuth = params.Settings.GetRatelimitServer().RateLimitBeforeAuth
```

Comment:
if ratelimit isn't configured at all, then this plugin will panic. the plugin should be able to initialize properly and immediately return during the `Process*` functions if the config is incomplete
```suggestion
		p.rateLimitBeforeAuth = params.Settings.GetRatelimitServer().GetRateLimitBeforeAuth()
```